lib/ddtrace/configuration/pin_setup.rb

```diff
@@ -0,0 +1,30 @@
+module Datadog
+  class Configuration
+    # PinSetup translates a flat hash into a Pin configuration
+    # This class should be removed if we ever remove/refactor the Pin class
+    class PinSetup
```

Comment:
Looks like this class is more or less a factory pattern for Pin. Is there something sufficiently complex/reusable that this class provides, but the Pin constructor doesn't make reasonably accessible?

Reply:
Yes, my first idea was to name this `PinFactory`. But I ended up concluding that the idea concept of a factory doesn't really apply since we're not responsible for instantiating `Pin` objects. Perhaps this is more like a "builder". What do you think?

Reply:
Oh, and answering to your second point, I don't think this should be merged to the `Pin` or be in the `Pin` namespace, since it's more an adapter if you will to the configuration API.